Brokerages/GDAX/GDAXBrokerage.cs

```diff
@@ -290,6 +295,125 @@ public override List<Cash> GetCashBalance()
+
+            if (request.Resolution == Resolution.Tick || request.Resolution == Resolution.Second)
+            {
+                Log.Error($"GDAXBrokerage.GetHistory(): {request.Resolution} resolution not supported, no history returned");
+                yield break;
+            }
+
+            Log.Trace($"GDAXBrokerage.GetHistory(): Submitting request: {request.Symbol.Value}: {request.Resolution} {request.StartTimeUtc} UTC -> {request.EndTimeUtc} UTC");
+
+            foreach (var tradeBar in GetHistoryFromCandles(request))
+            {
+                yield return tradeBar;
+            }
+        }
+
+        /// <summary>
+        /// Returns TradeBars from GDAX candles (only for Minute/Hour/Daily resolutions)
+        /// </summary>
+        /// <param name="request">The history request instance</param>
+        private IEnumerable<TradeBar> GetHistoryFromCandles(HistoryRequest request)
+        {
+            var productId = ConvertSymbol(request.Symbol);
+            var granularity = Convert.ToInt32(request.Resolution.ToTimeSpan().TotalSeconds);
+
+            var startTime = request.StartTimeUtc;
+            var endTime = request.EndTimeUtc;
+            var maximumRange = TimeSpan.FromSeconds(MaxDataPointsPerHistoricalRequest * granularity);
+
+            do
+            {
+                var maximumEndTime = startTime.Add(maximumRange);
+                if (endTime > maximumEndTime)
+                {
+                    endTime = maximumEndTime;
+                }
+
+                var restRequest = new RestRequest($"/products/{productId}/candles?start={startTime:o}&end={endTime:o}&granularity={granularity}", Method.GET);
+                var response = ExecuteRestRequest(restRequest, GdaxEndpointType.Public);
+
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    Log.Error($"GDAXBrokerage.GetHistory: request failed: [{(int)response.StatusCode}] {response.StatusDescription}, Content: {response.Content}, ErrorMessage: {response.ErrorMessage}");
+                    yield break;
+                }
+
+                var bars = ParseCandleData(request.Symbol, granularity, response.Content, startTime);
+
+                TradeBar lastPointReceived = null;
+                foreach (var datapoint in bars.OrderBy(x => x.Time))
+                {
+                    lastPointReceived = datapoint;
+                    yield return datapoint;
+                }
+
+                startTime = lastPointReceived?.EndTime ?? request.EndTimeUtc;
+                endTime = request.EndTimeUtc;
+            } while (startTime < request.EndTimeUtc);
+        }
+
+        /// <summary>
+        /// Parse TradeBars from JSON response
+        /// https://docs.pro.coinbase.com/#get-historic-rates
+        /// </summary>
+        private static IEnumerable<TradeBar> ParseCandleData(Symbol symbol, int granularity, string data, DateTime startTimeUtc)
+        {
+            if (data.Length > 0)
```

Comment:
style -- this is a guard clause, and they're typically written in inverse to reduce nesting:
```
if (data.Length == 0)
{
    yield break;
}
```